uta_tools/data_sources/seqrepo_access.py

```diff
@@ -1,68 +1,171 @@
 """A module for accessing SeqRepo."""
-from typing import Optional, List
+from typing import Optional, List, Tuple
 from biocommons.seqrepo import SeqRepo
+
 from uta_tools import SEQREPO_DATA_PATH, logger
 from os import environ
+from uta_tools.schemas import ResidueMode
 
 
 class SeqRepoAccess:
     """The SeqRepoAccess class."""
 
-    def __init__(self, seqrepo_data_path=SEQREPO_DATA_PATH):
+    def __init__(self, seqrepo_data_path: str = SEQREPO_DATA_PATH) -> None:
         """Initialize the SeqRepoAccess class.
         :param str seqrepo_data_path: The path to the seqrepo directory.
         """
         environ['SEQREPO_LRU_CACHE_MAXSIZE'] = "none"
-        self.seq_repo_client = SeqRepo(seqrepo_data_path)
+        self.seqrepo_client = SeqRepo(seqrepo_data_path)
+
+    def _get_start_end(
+            self, start: int, end: Optional[int] = None,
+            residue_mode: ResidueMode = ResidueMode.RESIDUE
+    ) -> Tuple[Optional[Tuple[int, int]], Optional[str]]:
+        """Get start and end in inter-residue (0-based) coords.
 
-    def get_sequence(self, transcript, start, end=None) -> Optional[str]:
-        """Return sequence for transcript at given positions.
-        :param str transcript: Accession
         :param int start: Start pos change
-        :param int end: End pos change
-        :return: Sequence
+        :param Optional[int] end: End pos change
+        :param ResidueMode residue_mode: Residue mode for start/end positions
+            Must be either `inter-residue` or `residue`
+        :return: start pos, end pos
         """
+        residue_mode = residue_mode.lower()
         if end is None:
-            end = start
+            end = start + 1
+        else:
+            if start == end:
+                end += 1
+
+        if residue_mode == ResidueMode.RESIDUE:
+            start -= 1
+            end -= 1
+
+        elif residue_mode != ResidueMode.INTER_RESIDUE:
+            return None, f"residue_mode must be either `inter-residue` or" \
+                         f" `residue`, not `{residue_mode}`"
+        return (start, end), None
+
+    def get_reference_sequence(
+            self, ac: str, start: int, end: Optional[int] = None,
+            residue_mode: ResidueMode = ResidueMode.RESIDUE
+    ) -> Tuple[Optional[str], Optional[str]]:
+        """Get reference sequence for transcript at given positions
+
+        :param str ac: Accession
+        :param int start: Start pos change
+        :param Optional[int] end: End pos change
+        :param ResidueMode residue_mode: Residue mode for start/end positions
+            Must be either `inter-residue` or `residue`
+        :return: Sequence at position, warning
+        """
+        sequence, warnings = self.check_sequence(
+            ac, start, end, residue_mode)
+        if warnings:
+            return None, warnings
+        else:
+            return sequence, None
 
+    def check_sequence(
+            self, ac: str, start: int = None,
```

Comment:
I believe the default value for `start` will cause a valueerror on line 34

Reply:
er TypeError

Reply:
Ope good catch. I don't think I meant to make it optional